### PR TITLE
[WASI-NN] Support more options for llama

### PR DESCRIPTION
**Pull request by dm4**
- Support `LLAMA_N_CTX` env for the context size
- Support `LLAMA_N_PREDICT` env for max tokens predicted
- Update ggml backend to b1273

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: plugins/wasi_nn/ggml.cpp [hunk 1]
```diff
@@ -12,6 +12,20 @@
 
 namespace WasmEdge::Host::WASINN::GGML {
 #ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_GGML
+llama_context_params wasmedge_llama_context_params() noexcept {
+  llama_context_params Params = llama_context_default_params();
+  const char *LlamaNContextEnv = std::getenv("LLAMA_N_CTX");
+  const char *LlamaLogEnv = std::getenv("LLAMA_LOG");
+  if (LlamaNContextEnv != nullptr) {
+    Params.n_ctx = std::stoi(LlamaNContextEnv);
```
Comment (hydai) [inline plugins/wasi_nn/ggml.cpp:20]: Should we cover these conversions in a try block? If the given n_ctx is invalid, what will happen?

File: plugins/wasi_nn/ggml.cpp [hunk 2]
```diff
@@ -138,8 +152,17 @@ Expect<ErrNo> compute(WasiNNEnvironment &Env, uint32_t ContextId) noexcept {
   // TODO: recompute a compressed context based on previous tokens once the
   // cache is full.
   const int MaxContextSize = llama_n_ctx(GraphRef.LlamaContext);
+  int NPredict = std::numeric_limits<int>::max();
+  const char *LlamaNPredictEnv = std::getenv("LLAMA_N_PREDICT");
+  if (LlamaNPredictEnv != nullptr) {
+    NPredict = std::stoi(LlamaNPredictEnv);
```
Comment (hydai) [inline plugins/wasi_nn/ggml.cpp:158]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasi_nn/ggml.cpp (hunk 1 above)
Comment (dm4) [reply plugins/wasi_nn/ggml.cpp]: Added.

File: plugins/wasi_nn/ggml.cpp (hunk 2 above)
Comment (dm4) [reply plugins/wasi_nn/ggml.cpp]: Added.

**Review by hydai — APPROVED** [review]

**Pull request merged**